code/game/machinery/doors/door.dm

```diff
@@ -270,7 +270,7 @@
 
 /obj/machinery/door/proc/autoclose()
 	var/obj/machinery/door/airlock/A = src
-	if(!A.density && !A.operating && !A.locked && !A.welded && A.autoclose)
+	if(!qdeleted(src) && A && !A.density && !A.operating && !A.locked && !A.welded && A.autoclose)

```

Comment:
```
&& A &&
```
`A` is `src`. It cannot be null, or this proc call would be impossible.

... also, why the hell does this typecast `src` to a different type?! That's either completely unnecessary, or a runtime waiting to happen.